test/integration/self-provider-staking.test.ts

```diff
@@ -0,0 +1,586 @@
+    })
+
+    it('success - get min stake amount', async () => {
+      expect(provider.contract.MIN_STAKE_AMOUNT).toBe(
+        BigInt('1000000000000000000000')
+      )
+    })
+  })
+
+  describe('Admin Functions', () => {
+    describe('pause()', () => {
+      it('failure - the signer is not admin', async () => {
+        expect.assertions(1)
+
+        await expect(notAdminProvider.contract.pause()).rejects.toThrow(
+          commonErrorMessage.addressIsNotAdmin
+        )
+      })
+
+      it('success - pause contract', async () => {
+        expect.assertions(1)
+        const tx = await provider.contract.pause()
+
+        await waitUntilConfirmationCompleted(
+          provider.contract.jsonRpcProvider,
+          tx
+        )
+
+        const isPaused = await provider.contract.isPaused()
+
+        expect(isPaused).toBe(true)
+      })
+    })
+
+    describe('unpause()', () => {
+      it('failure - the signer is not admin', async () => {
+        expect.assertions(1)
+
+        await expect(notAdminProvider.contract.unpause()).rejects.toThrow(
+          commonErrorMessage.addressIsNotAdmin
+        )
+      })
+
+      it('success - unpause the contract', async () => {
+        expect.assertions(1)
+
+        const tx = await provider.contract.unpause()
+
+        await waitUntilConfirmationCompleted(
+          provider.contract.jsonRpcProvider,
+          tx
+        )
+
+        const isPaused = await provider.contract.isPaused()
+
+        expect(isPaused).toBe(false)
+      })
+    })
+  })
+
+  describe('Functions', () => {
+    describe('addPlan()', () => {
+      it('failure - not an admin', async () => {
+        const res = notAdminProvider.contract.addPlan(
+          BigInt(180 * 24 * 60 * 60),
+          BigInt(25)
+        )
+
+        await expect(res).rejects.toThrow(commonErrorMessage.addressIsNotAdmin)
+      })
+
+      it('failure - lockDuration is less than 0', async () => {
+        const res = provider.contract.addPlan(BigInt(-1), BigInt(25))
+
+        await expect(res).rejects.toThrow(
+          stakingErrorMessage.nonNegativeLockDuration
+        )
+      })
+
+      it('failure - apy is less than 0', async () => {
+        const res = provider.contract.addPlan(BigInt(0), BigInt(-1))
+
+        await expect(res).rejects.toThrow(stakingErrorMessage.nonNegativeApy)
+      })
+
+      it('success - the plan is registered', async () => {
+        const res = provider.contract.addPlan(
+          BigInt(180 * 24 * 60 * 60),
+          BigInt(25)
+        )
+
+        await expect(res).resolves.not.toThrow()
+      })
+    })
+
+    describe('stakingEndTime()', () => {
+      it('success - get staking end time', async () => {
+        const res = await provider.contract.stakingEndTime()
+
+        expect(typeof res).toBe('bigint')
+      })
+    })
+
+    describe('stake()', () => {
+      it('failure - Plan Id does not exist', async () => {
```

Comment:
The test description must be `Contract is paused`

Reply:
Nice catch, thanks.